### PR TITLE
Rc display all grade sheets

### PR DESCRIPTION
**Pull request by robert-czwartosz**
Hey, displaying all grade sheets is done.
You can add empty grade sheet and delete selected grade sheets.
Component is in storybook and in admin dashboard on dev server.

### CODE REVIEW

File: frontend/src/api/Project.service.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        private api = new BaseService(),
+        private sectionApi = new SectionService()
+    ) { }
+
+    async getProject(id: string): Promise<Project | null> {
+        let projectRes: AxiosResponse<ProjectData>;
+        let project: ProjectData;
+        try {
+            projectRes = await this.api.get(`/projects/${id}`);
+            console.log ('============================================================')
```
Comment (kjreczek) [inline frontend/src/api/Project.service.ts:19]: is console.log necessary?

File: frontend/src/api/Project.service.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            project = projectRes.data;
+            console.log(project)
```
Comment (kjreczek) [inline frontend/src/api/Project.service.ts:21]: is console.log necessary?

File: frontend/src/api/Section.service.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    
+    constructor(private api = new BaseService()) { }
+
+    async getSection(id: string): Promise<Section | null> {
+        let sectionRes: AxiosResponse<SectionData>;
+        let section: SectionData;
+        try {
+            sectionRes = await this.api.get(`/sections/${id}`);
+            section = sectionRes.data;
+            console.log(section);
```
Comment (kjreczek) [inline frontend/src/api/Section.service.ts:16]: is console.log necessary?

File: frontend/src/api/Sheet.service.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        });
+    }
+
+    deleteSheet = async (id: string) => {
+        await this.api.delete('/grade/sheets/' + id);
+    }
+
+    getMentorSheets = async (mentorId?: string): Promise<GradeSheetData[] | null>  => {
+        let gradeSheetsRes;
+        console.log({mentorId});
```
Comment (kjreczek) [inline frontend/src/api/Sheet.service.ts:61]: is console.log necessary?

File: frontend/src/api/Team.service.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            surname: team?.mentor?.surname ?? '---',
+            courseName: course.name,
+            users: team.users,
+        }) );
+    };
+
+    getTeam = async (id: string): Promise<TeamInfo> => {
+        const teamRes = await this.api.get('/teams/' + id);
+        const team = teamRes.data as TeamData;
+        console.log(team);
```
Comment (kjreczek) [inline frontend/src/api/Team.service.ts:33]: is console.log necessary?

File: frontend/src/api/Team.service.ts [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+                surname: user.surname,
+                type: userTypeDict[user.type],
+                status: userStatusDict[user.status],
+                averageGrade,
+                maxGrade,
+            }
+        });
+        teamInfo.teamAvgGrade = teamInfo.users
+            .reduce((acc, user) => user.averageGrade ?? 0 + acc, 0) / teamInfo.users.length;
+        console.log(teamInfo);
```
Comment (kjreczek) [inline frontend/src/api/Team.service.ts:74]: is console.log necessary?

File: frontend/src/api/TeamProject.service.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    constructor(
+        private api = new BaseService(),
+        private projectApi = new ProjectService(),
+        private sheetApi = new SheetService()
+    ) { }
+
+    getTeamProjects = async (teamId: string, mentorId?: string): Promise<TeamProject[]> => {
+        const projectsRes = await this.api.get(`/teams/${teamId}/projects`);
+        const projectsData = projectsRes.data as TeamProjectData[];
+        console.log(projectsRes);
```
Comment (kjreczek) [inline frontend/src/api/TeamProject.service.ts:17]: is console.log necessary?

File: frontend/src/api/TeamProject.service.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+            return {
+                id: project._id,
+                name: project.projectName,
+                overallGrade: 0,
+                sectionName: parentProject?.sectionName ?? '---',
+                url: project.projectUrl,
+                description: project.description,
+        }} ) );
+        const grades = await this.sheetApi.getMentorSheets(mentorId);
+        console.log(grades);
```
Comment (kjreczek) [inline frontend/src/api/TeamProject.service.ts:29]: is console.log necessary?

File: frontend/src/api/User.service.ts [hunk 9]
```diff
@@ -13,6 +18,43 @@ export default class UserService {
...
+        const users = await this.getUsersOfType('Participant');
+        const teams = await this.teamsApi.getTeams();
+        const result = users.filter(user => {
+            return teams.every( team => !(team.users) || team.users.every(u => {
+                if(u._id !== user.id)
+                    console.log('asd');
+                return u._id !== user.id;
+            }) );
+        });
+        console.log({result});
```
Comment (kjreczek) [inline frontend/src/api/User.service.ts:56]: is console.log necessary?

**Review by kjreczek — APPROVED** [review]
Everything looks good, I left only comments about console.log.

**Review by fszwedo — APPROVED** [review]
looks good to me, there are a few conflicts unfortunately due to recent mergers

right now courseid is being set in the App.tsx, but this probably will be changed with the conflict resolving

**Pull request merged**